### PR TITLE
[Merged by Bors] - chore(GroupTheory/GroupAction/Group): Delete

### PR DESCRIPTION
**Pull request by YaelDillies**
All lemmas can be moved to earlier files

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Algebra/GroupWithZero/Action/Defs.lean
Comment (jcommelin) [inline Mathlib/Algebra/GroupWithZero/Action/Defs.lean:8]: This seems to amount to a 12% increase of the number transitive imports...

Since this is a `Defs` file, I would rather prefer to see the number go down.

Comment (YaelDillies) [reply Mathlib/Algebra/GroupWithZero/Action/Defs.lean:8]: It is a very recent `Defs` file, however. I am still working on it. Here are my future plans:
1. Move some of the content from `GroupTheory.GroupAction.Group` to this file (this PR)
2. Move most of the content of `Algebra.SMulWithZero` to this file (not yet opened, but you can see the result in #13027)
3. Think about a nice way to split things into `Defs` and `Basic`

I can't really do the last step before the first two, so I would rather stick to my plan and declare this temporary 12% increase as fine.

Comment (j-loreaux) [reply Mathlib/Algebra/GroupWithZero/Action/Defs.lean:8]: Why can't you do (2) first and then (1) and (3) simultaneously?

Comment (YaelDillies) [reply Mathlib/Algebra/GroupWithZero/Action/Defs.lean:8]: (2) will increase imports as well

**Comment by j-loreaux** [comment]
@jcommelin I'll let you decide, since you initially raised the objection. I'm content with Yaël's indication to later split the file into `Defs` + `Basic` in some helpful way.

**Comment by jcommelin** [comment]
Ok, let's go ahead.

bors merge

**Pull request closed without merging**
